sdk/core/azure-core/src/main/java/com/azure/core/implementation/logging/DefaultLogger.java

```diff
@@ -59,14 +59,20 @@ public DefaultLogger(Class<?> clazz) {
      * name passes in.
      */
     public DefaultLogger(String className) {
-        String classPath;
+        this(getClassPathFromClassName(className), false);
+    }
+
+    private static String getClassPathFromClassName(String className) {
         try {
-            classPath = Class.forName(className).getCanonicalName();
+            return Class.forName(className).getCanonicalName();
```

Comment:
nit: maybe call overload with `initialize` flag set to false to minimize side-effects?

Reply:
I'm a bit confused on this comment, this is a utility method to help the `String`-based constructor as `this` has to be the first statement in a constructor.